### PR TITLE
Implement a build artifact archiving mechanism

### LINKED ISSUE #144: Expose the build workspace over http
To be able to retrieve the build artifacts or view log files.

Since every variant has its own workspace, this feature would require some changes to how things work in bazooka:

Right now, we perform a single checkout in a volume, and then `ADD` the workspace to the variants images.

To be able to serve the workspaces after the build, we would need to either:

* Make a copy of the checkout for every variant and mount the copy as a volume in the variant image
* Or let the variant image copy its embedded workspace in a volume at the build end

I haven't yet thought about the pros/cons of these alternatives.

We also need to be careful of what happens with large workspaces: duplicating a multi-hundred megabytes workspace for N variants could quickly get out of hand.

### PR DESCRIPTION
**Pull request by jawher**
This feature relies on the following configuration keys:

* archive: takes a single or a list of globs. The matching files are archived regardless of the result of the build
* archive_success: takes a single or a list of globs. The matching files are archived in case of a successfull build
* archive_failure: takes a single or a list of globs. The matching files are archived in case of a failed build

The archived files are then exposed over http using the following endpoint:

bzk_server:3000/variant/{id}/artifacts/{path}

Where id is the variant id and path the path to an archived artifact.

Fixes #144

### CODE REVIEW

**Comment by jawher** [comment]
An example config for a java project:

```yaml
language: java
jdk:
  - oraclejdk8
  - openjdk8
  - oraclejdk7
archive:
  - target/classes
archive_success: target/*.jar
archive_failure: 
  - target/surefire-reports
```

File: orchestration/runner.go [hunk 1]
```diff
@@ -110,7 +114,7 @@ func (r *Runner) runContainer(logger Logger, vd *variantData, env map[string]str
 	}
 	if err = container.Remove(&docker.RemoveOptions{
 		Force:         true,
-		RemoveVolumes: true,
+		RemoveVolumes: false,
```
Comment (julienvey) [inline orchestration/runner.go]: Any reason to keep volumes ?

Comment (jawher) [inline orchestration/runner.go]: None :)

**Comment by jawher** [comment]
Just a note: the approach taken here, i.e. storing and serving the build artifacts from the filesystem, will be a problem for #71: distributed build since this pins a particular build to a particular machine.

**Pull request merged**